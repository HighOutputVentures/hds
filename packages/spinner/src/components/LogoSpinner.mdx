import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import StorybookHeader from '../../../../components/StorybookHeader';

<Meta title="Components/Spinner/LogoSpinner" />

<StorybookHeader
  title="HDS Spinner"
  command="npm install @highoutput/hds-spinner"
/>

<Canvas>
  <Story id="components-spinner-logospinner--default"></Story>
</Canvas>

<ArgsTable />
