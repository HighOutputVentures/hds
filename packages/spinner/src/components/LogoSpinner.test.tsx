import { cleanup, render } from '@testing-library/react';
import * as React from 'react';
import renderer from 'react-test-renderer';
import LogoSpinner from './LogoSpinner';

const duration = Math.random();

jest.mock('./LogoSpinner', () => {
  return ({ children }: React.PropsWithChildren<{ duration: number }>) => {
    return <div>{children}</div>;
  };
});

describe('LogoSpinner', () => {
  afterEach(cleanup);

  it('Should render children', () => {
    const { getByText } = render(
      <LogoSpinner duration={duration}>
        <div>Hello world</div>
      </LogoSpinner>
    );

    expect(getByText('Hello world')).toBeDefined();
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <LogoSpinner duration={duration}>
        <div>Hello world</div>
      </LogoSpinner>
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });
});
