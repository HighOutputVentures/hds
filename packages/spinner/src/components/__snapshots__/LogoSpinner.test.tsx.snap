// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSpinner Should match snapshot 1`] = `
<div>
  <div>
    Hello world
  </div>
</div>
`;
